Remove const_transmute feature dependency

RSSchermer opened the pull request:
`const_transmute` seems to have been stabilized on current nightly, see https://github.com/rust-lang/rust/pull/72920. This caused an unrecognized feature flag error when compiling against nightly.

RalfJung commented:
The feature flag will still be needed when using `offset_of!` in a `const fn`, though. That should be documented at least.

RSSchermer commented:
Ah, I hadn't noticed yet that the flag essentially got replaced with `const_fn_transmute`. I added a test case for `offset_of!` inside a `const fn` that seemed to indicate the `const_fn` feature flag is also requirement. I've added a line to the readme.

RalfJung commented:
This is great, thanks. :)
Can you run rustfmt on it so that CI passes?

RSSchermer commented:
Ah, sorry, silly oversight.
